snuba/utils/schemas.py

```diff
@@ -558,3 +557,69 @@ def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
 
     def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
+
+
+class Tuple(ColumnType[TModifiers]):
+    def __init__(
+        self,
+        types: tuple[ColumnType[TModifiers], ...],
+        modifiers: Optional[TModifiers] = None,
+    ) -> None:
+        super().__init__(modifiers)
+        self.types = types
+
+    def _repr_content(self) -> str:
+        return ", ".join("{}".format(repr(v)) for v in self.types)
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            self.__class__ == other.__class__
+            and self.get_modifiers() == cast(Enum[TModifiers], other).get_modifiers()
+            and self.types == cast(Tuple[TModifiers], other).types
+        )
+
+    def _for_schema_impl(self) -> str:
+        return "Tuple({})".format(", ".join([repr(t) for t in self.types]))
+
+    def set_modifiers(self, modifiers: Optional[TModifiers]) -> Tuple[TModifiers]:
+        return Tuple(types=self.types, modifiers=modifiers)
+
+    def get_raw(self) -> Tuple[TModifiers]:
+        return Tuple(self.types)
+
+
+class NamedTuple(ColumnType[TModifiers]):
+    def __init__(
+        self,
+        types: tuple[tuple[str, ColumnType[TModifiers]], ...],
+        modifiers: Optional[TModifiers] = None,
+    ) -> None:
+        super().__init__(modifiers)
+        self.types = tuple(Column(*t) for t in types)
+
+    def flatten(self, name: str) -> Sequence[FlattenedColumn]:
+        return [
+            FlattenedColumn(name, column.name, Array(column.type))
+            for column in self.types
+        ]
```

Comment:
I don't think you need this, why changing from the default implementation?

Reply:
Not quite sure, I thought it made sense given how it was done in `Nested` but I can't exactly remember why so it's probably not useful. Will remove.